src/fft4g32.c

```diff
@@ -9,11 +9,12 @@
 
 #if WITH_CR32
 #include "rdft_t.h"
-static void * null(void) {return 0;}
-static void forward (int length, void * setup, double * H) {lsx_safe_rdft_f(length,  1, H); (void)setup;}
-static void backward(int length, void * setup, double * H) {lsx_safe_rdft_f(length, -1, H); (void)setup;}
+static void * null(int length) {return 0; (void)length;}
+static void nothing(void * setup) {(void)setup;}
+static void forward(int length, void * setup, void * H, void * scratch) {lsx_safe_rdft_f(length, 1, (double * ) H); (void)setup; (void)scratch;}
```

Comment:
Can you test this works?
```suggestion
static void forward(int length, void * setup, double * H, void * scratch) {lsx_safe_rdft_f(length, 1, H); (void)setup; (void)scratch;}
```

Reply:
It might work, but I don't think it is correct. As far as I know, the signature in the function declaration used during the function call (from `rdft_t.h`) and the signature in the function definition have to be identical. For example, this code will not compile with gcc:

```c
// Declaration (like in rdft_t.h)
void forward(void * H);

// Definition (like in fft4g32.c)
void forward(double * H) {}
```

Reply:
Yes, it's idiomatically not right. But in this case, avfft*.c and pffft*.c should be modified in same manner.
I want to minimize modifications to libsoxr, so I'm asking this way of changes.
I think it's likely to work anyway.

Reply:
Yeah, I don't think there is any platform where the destination type of a pointer matters. I changed it back.